packaging/rpm/scripts/post-install.sh

```diff
@@ -6,10 +7,41 @@ set -e
 #   * on upgrade: $1 > 1
 # https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#_syntax
 
-# from %post in st2-packages.git/packages/st2/rpm/st2.spec
-%service_post st2actionrunner st2api st2stream st2auth st2notifier st2workflowengine
-%service_post st2rulesengine st2timersengine st2sensorcontainer st2garbagecollector
-%service_post st2scheduler
+_ST2_SERVICES="
+st2actionrunner
+st2api
+st2auth
+st2garbagecollector
+st2notifier
+st2rulesengine
+st2scheduler
+st2sensorcontainer
+st2stream
+st2timersengine
+st2workflowengine
+"
 
-# make sure that our socket generators run
-systemctl daemon-reload >/dev/null 2>&1 || true
+# Native .rpm specs use macros that get expanded into shell snippets.
+# We are using nfpm, so we inline the macro expansion here.
+# %systemd_post
+#   EL8: https://github.com/systemd/systemd/blob/v239/src/core/macros.systemd.in
+#   EL9: https://github.com/systemd/systemd/blob/v252/src/rpm/macros.systemd.in
+
+if [ $1 -eq 1 ]; then
+    # Initial installation
+    if [ -x "/usr/lib/systemd/systemd-update-helper" ]; then # EL 9
+        /usr/lib/systemd/systemd-update-helper install-system-units ${_ST2_SERVICES} || :
+    else # EL 8
+        systemctl --no-reload preset ${_ST2_SERVICES} &>/dev/null || :
+    fi
+fi
+
+# TODO: Maybe remove this as 'preset' (on install above) enables units by default
+systemctl --no-reload enable ${_ST2_SERVICES} &>/dev/null || :
```

Comment:
I debated removing this now. I left a TODO for now. I think we should wait to remove it until we can actually test the built rpms.